### PR TITLE
Attempt at porting to zio2

### PR DESCRIPTION
**Pull request by sirocchj**
I dunno 🤷‍♂️

### CODE REVIEW

File: zio/src/test/scala/ConsoleLogWriterTest.scala [hunk 1]
```diff
@@ -260,5 +264,6 @@ final class ConsoleLogWriterTest extends AnyWordSpecLike with Matchers with zio.
     }
   }
 
-  override def run(args: List[String]): ZIO[ZEnv, Nothing, ExitCode] = IO.succeed(ExitCode.success)
+  override def run: ZIO[Environment with ZIOAppArgs with Scope, Any, Any] =
```
Comment (barambani) [inline zio/src/test/scala/ConsoleLogWriterTest.scala:267]: Would this work ?
```suggestion
  override def run: ZIO[Environment with ZIOAppArgs with Scope, Nothing, ExitCode] =
```

File: zio/src/test/scala/ReadmeRioConstructionCodeSnippetsTest.scala [hunk 2]
```diff
@@ -43,60 +43,75 @@ import scala.annotation.nowarn
...
-      val log4sCase1: RLayer[Has[AConfig], ZLogWriter] =
-        ZLayer.fromServiceM { config =>
-          log4sFromName.provide(config.logName)
-        }
-      val scribe4sCase1: RLayer[Has[AConfig], ZLogWriter] =
-        ZLayer.fromServiceM { config =>
-          scribeFromName.provide(config.logName)
-        }
+      val log4sCase1: RLayer[AConfig, ZLogWriter] = ZLayer {
+        ZIO.service[AConfig].flatMap(c => log4sFromName.provideEnvironment(ZEnvironment(c.logName)))
```
Comment (barambani) [inline zio/src/test/scala/ReadmeRioConstructionCodeSnippetsTest.scala:47]: serviceWithZio could work here
```suggestion
        ZIO.serviceWithZIO(c => log4sFromName.provideEnvironment(ZEnvironment(c.logName)))
```

**Review by barambani — APPROVED** [review]

**Review by barambani — APPROVED** [review]

File: zio/src/test/scala/ConsoleLogWriterTest.scala [hunk 3]
```diff
@@ -231,7 +235,7 @@ final class ConsoleLogWriterTest extends AnyWordSpecLike with Matchers with zio.
     "created with level Trace" should {
       "log all the messages" in {
         val out = capturedConsoleOutOf {
-          val cl = consoleLogUpToLevel(Trace)
+          val cl = consoleLogUpToLevel(log.effect.LogLevels.Trace)
```
Comment (sirocchj) [inline zio/src/test/scala/ConsoleLogWriterTest.scala:238]: for the record, no idea why this was needed, but not in all other cases...

Comment (barambani) [reply zio/src/test/scala/ConsoleLogWriterTest.scala:238]: It's ambiguous with zio.Trace I guess.

Comment (sirocchj) [reply zio/src/test/scala/ConsoleLogWriterTest.scala:238]: Yeah, you're absolutely right.. I'll create a mini-PR to hide it

Comment (sirocchj) [reply zio/src/test/scala/ConsoleLogWriterTest.scala:238]: #530

**Pull request merged**